### DISCUSSION TITLE
Add more options to the clipboard history or add an API to extend it

### DISCUSSION THREAD

**Discussion by danielo515** [johnlindquist/kit#1090] [Ideas]
Most people don't understand the time saver that a proper clipboard history is. I do, and I use it extensively. However, a lot of times I want to do even more than just going to an element and paste it. The list of things I wish to be able to do is not long, but it is not trivial either. 

- Sometimes I want to edit an item before pasting
- Sometimes I want to paste it as plain text, not as rich text
- Often times I want to store them as a snippet
- I want to pin some entries
- I wish I could add some labels to better organise them
- I really like having a fuzzy finder, that takes each space as a separate query, so I can match individual sections of the entry

I think this will make sense for more people ( I hope)

**Comment by johnlindquist** [comment]
You can use the following APIs in a script:

```
await getClipboardHistory()
await removeClipboardItem()
await clearClipboardHistory()
```

While everything you're asking for is definitely possible, I won't have the bandwidth to build out a full-featured clipboard manager for a while.

**Reply by danielo515** [reply]
Will you be open to take a PR that adds some of them?

**Reply by johnlindquist** [reply]
Yup, always open to PRs ❤️
